api/v1beta1/nutanixcluster_types.go

```diff
@@ -91,6 +99,39 @@ type NutanixCluster struct {
 	Status NutanixClusterStatus `json:"status,omitempty"`
 }
 
+// NutanixFailureDomain configures failure domain information for Nutanix.
+type NutanixFailureDomain struct {
+	// name defines the unique name of a failure domain.
+	// Name is required and must be at most 64 characters in length.
+	// It must consist of only lower case alphanumeric characters and hyphens (-).
+	// It must start and end with an alphanumeric character.
+	// This value is arbitrary and is used to identify the failure domain within the platform.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=`[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+	Name string `json:"name"`
+
+	// cluster is to identify the cluster (the Prism Element under management of the Prism Central),
+	// in which the Machine's VM will be created. The cluster identifier (uuid or name) can be obtained
+	// from the Prism Central console or using the prism_central API.
+	// +kubebuilder:validation:Required
+	Cluster NutanixResourceIdentifier `json:"cluster"`
```

Comment:
per docs, <pre>
failureDomains - is a FailureDomains type indicating the failure domains that machines should be placed in. FailureDomains is a map, defined as map[string]FailureDomainSpec. A unique key must be used for each FailureDomainSpec. FailureDomainSpec is defined as:
controlPlane (bool): indicates if failure domain is appropriate for running control plane instances.
attributes (map[string]string): arbitrary attributes for users to apply to a failure domain.
</pre>
should we define a map[string]string for attributes instead of above

Reply:
These attributes are on the status of the object, not on the spec. Hence we cannot set them.